blueprints/in-repo-addon/index.js

```diff
@@ -31,5 +32,47 @@ module.exports = {
     }
 
     fs.writeFileSync(packagePath, JSON.stringify(contents, null, 2));
+  },
+
+  afterUninstall: function(options) {
+    var packagePath = path.join(this.project.root, 'package.json');
+    var contents    = fs.readJsonSync(packagePath);
+    var name        = stringUtil.dasherize(options.entity.name);
+    var newPath     = ['lib', name].join('/');
+    var paths;
+    var newPathIndex;
+
+    contents['ember-addon'] = contents['ember-addon'] || {};
+    paths = contents['ember-addon']['paths'] = contents['ember-addon']['paths'] || [];
+    newPathIndex = paths.indexOf(newPath);
+
+    if (newPathIndex > -1) {
+      paths.splice(newPathIndex, 1);
+      if (paths.length === 0) {
+        delete contents['ember-addon']['paths'];
+      }
+    }
+
+    try {
+      if (fs.lstatSync(newPath).isDirectory()) {
+        fs.rmdirSync(newPath);
```

Comment:
yes, knowing if a dir can be removed or not is a surprisingly hard problem. I don't think at this point their is sufficient information to know if it can or can't be removed. We may want to leave it.